internal/core/data/device_test.go

```diff
@@ -52,16 +55,20 @@ func TestCheckMaxLimitOverLimit(t *testing.T) {
 
 	testedLimit := math.MaxInt32
 
-	expectedErr := checkMaxLimit(testedLimit)
+	expectedErr := checkMaxLimit(testedLimit, mockLoggingClient)
 
 	if expectedErr == nil {
 		t.Errorf("Exceeded limit and should throw error")
 	}
 }
 
 func TestMain(m *testing.M) {
-	testRoutes = LoadRestRoutes()
-	LoggingClient = logger.NewMockClient()
+	dic := di.NewContainer(di.ServiceConstructorMap{
+		container.LoggingClientInterfaceName: func(get di.Get) interface{} {
+			return mockLoggingClient
+		}})
+
+	testRoutes = LoadRestRoutes(dic)
```

Comment:
Per our side conversation, I question what purpose this test serves beyond code coverage.  It doesn't appear to be actually testing anything.  If that's the case, it should be removed.

As a more general statement, there's something amiss if the dependency injection container is required for a unit test.

Reply:
These tests are a bit messed up. I removed this test, which doesn't look like it would affect anything, and other tests started to fail due to panics. I will need to look deeper into this and restructure these tests.

Reply:
It turns out that this test was instanciating the global variables:

- mdc (meta-data client)
- msgClient
- chEvents 

These two tests were expecting those variables to be populated, but were not doing the population themselves:

https://github.com/edgexfoundry/edgex-go/blob/dc82d8d40a8e1e3d622a5a3b2b216fa3e9cf0243/internal/core/data/valuedescriptor_test.go#L314

https://github.com/edgexfoundry/edgex-go/blob/dc82d8d40a8e1e3d622a5a3b2b216fa3e9cf0243/internal/core/data/event_test.go#L210

So I updated them to populate the global variables they need to execute the test. These will go away with my future work to eliminate the global variables.

Reply:
@AnthonyMBonafide great illustration of why tests should be self-contained.  The hidden dependency on these global variables being set in different tests would be hard to troubleshoot/find if the tests that depended on them failed because of related changes.